### DISCUSSION TITLE
Installing on Ubuntu

### DISCUSSION THREAD

**Discussion by cskumar6381** [aws/credentials-fetcher#65] [Q&A]
Trying to get this on Ubuntu 22.04. Was able to get gRPC compiled and installed with all the modules. cmake is failing to compile it. 
[error.txt](https://github.com/aws/credentials-fetcher/files/12512217/error.txt)

Thanks in advance.

**Comment by smhmhmd** [comment]
Hi @cskumar6381 

grpc (especially absl) needs c++17, some new types got added.
This is being worked on here - PR https://github.com/aws/credentials-fetcher/pull/64

**Reply by smhmhmd** [reply]
Hi @cskumar6381 
Building on Ubuntu is now available at https://github.com/aws/credentials-fetcher/blob/mainline/docker/Dockerfile-ubuntu-20.04

**Reply by cskumar6381** [reply]
Thanks for the update. Tried the steps on a Ubuntu system.  Getting stuck with errors (snippet attaced)
![Screenshot 2023-11-15 205348](https://github.com/aws/credentials-fetcher/assets/144009421/72163bb2-071b-4e87-95a3-2027595fc93d)

**Comment by saikiranakula-amzn** [comment]
Thank you cskumar for reaching out to us. We will take a look at this and keep you posted on the next steps

**Comment by saikiranakula-amzn** [comment]
Looking at the error messages, you seem you are missing the installation of jsoncpp package. Can you please install "apt-get install jsoncpp" and see if it resolved the issue you are facing?

**Reply by cskumar6381** [reply]
Thanks, Sai, for quick response.  Please find the attached output snippet.  FYI this is Ubuntu 22.04 
![Screenshot 2023-11-16 094319](https://github.com/aws/credentials-fetcher/assets/144009421/245ac3b7-27e3-4a80-aa49-4d51b30e515f)
![Screenshot 2023-11-16 095521](https://github.com/aws/credentials-fetcher/assets/144009421/214aefde-7439-4ad0-8334-76506f2f9da9)

**Reply by cskumar6381** [reply]
looks like its seraching for json.h in /usr/include  
yet the actual file is /usr/include/jsoncpp/json/   (in Ubuntu) .. So did "ln -s /usr/include/jsoncpp/json/ /usr/include/json" to fix this error.

Instead of #include <json/json.h>, it's #include <jsoncpp/json/json.h> .

But now i see a different error. (snippet attached) 
![Screenshot 2023-11-16 102758](https://github.com/aws/credentials-fetcher/assets/144009421/d2534aa9-24f3-4b8e-9839-36959301c0b4)

**Reply by saikiranakula-amzn** [reply]
Looking at this, this seems to be an error on arguments that are getting passed. Can you please provide your email. We can have a short call to get you unblocked.

**Reply by cskumar6381** [reply]
ch.v.sharat@gmail.com

**Reply by saikiranakula-amzn** [reply]
Thank you Sharath. We will send out meeting invite to triage the issue.

**Comment by smhmhmd** [comment]
Thanks for the discussion yesterday, please let us know if you need further help.
